.clang-format

```diff
@@ -0,0 +1,184 @@
+# These options seem to change all the time.  I've removed things that are C++,
+# Objective-C, or JSON specific.  For reference:
+# https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+
+---
+Language:        Cpp
+BasedOnStyle: GNU
+AccessModifierOffset: -2
+AlignAfterOpenBracket: Align
+AlignArrayOfStructures: None
```

Comment:
If it matters, some of these options require particular minimum clang-format versions. For example, this one requires clang-format 13. The stakes are low, of course. Worst case scenario, this little-used developer make target breaks.

Reply:
Yes, that's true.  In #3829, I brought up the fact that cppcheck changes what it can find and what needs to be suppressed over time too.  I'm not sure if we've ever had any guidelines on what versions of these tools we're using and supporting.  Given that it's all developer stuff, I lean towards not worrying about it too much.  I expect that we're all using fairly modern stuff.